pkg/appStore/installedApp/service/AppStoreDeploymentService.go

```diff
@@ -492,12 +492,50 @@ func (impl *AppStoreDeploymentServiceImpl) RollbackApplication(ctx context.Conte
 	}
 	// Rollback starts
 	var success bool
-	if util2.IsHelmApp(installedApp.AppOfferingMode) {
+	//case when the request is for external app case
+	if installedApp.Id == 0 {
 		installedApp, success, err = impl.eaModeDeploymentService.RollbackRelease(ctx, installedApp, request.GetVersion(), tx)
 		if err != nil {
 			impl.logger.Errorw("error while rollback helm release", "error", err)
 			return false, err
 		}
+	} else if installedApp.DeploymentAppType == util.PIPELINE_DEPLOYMENT_TYPE_HELM && installedApp.Id > 0 {
```

Comment:
use util.IsHelmApp(deploymentConfig.DeploymentAppType) and remove unnecessary check installedApp.Id > 0